fields/sql.go

```diff
@@ -186,10 +188,12 @@ func (s *Scanner) Scan(src interface{}) error {
 			return nil
 		}
 	case *time.Time:
-		if _, ok := src.(time.Time); ok {
-			s.value.Set(reflect.ValueOf(src))
-			return nil
+		t := mysql.NullTime{}
+		if err := t.Scan(src); err != nil {
+			return err
 		}
+		s.value.Set(reflect.ValueOf(t.Time))
```

Comment:
i'm surprised we need to call into mysql driver here?

Reply:
We _could_ handle the various permutations of how time is stored ourselves to decouple this from the mysql driver.

Reply:
I don't think this scans nil correctly? We should check t.Valid and set s.isValid accordingly.

Reply:
Null is already handled on line 170

Reply:
Can we have a test case that shows we scan nil MySQL time into nil time.Time correctly?

Reply:
(non-pointer converts into zero value)